src/api.ts

```diff
@@ -397,34 +408,94 @@ function createApi({
 
       const { port1, port2 } = new MessageChannel()
 
-      port2.on('message', (/** message: PortMessage */) => {
-        // TODO: do something with progress messages
-      })
+      activeImports.set(importId, port2)
+
+      return new Promise((res, rej) => {
+        // Initially use a longer duration to account for worker startup
+        let timeoutId = createTimeout(10000)
+
+        port2.on('message', handleFirstProgressMessage)
+        port2.on('message', resetTimeout)
+
+        // Can use a normal event emitter that emits an `abort` event as the abort signaler for Piscina,
+        // which allows us to not have to worry about globals or relying on polyfills
+        // https://github.com/piscinajs/piscina#cancelable-tasks
+        const abortSignaler = new EventEmitter()
+
+        piscina
+          .run(
+            {
+              dbPath: db.name,
+              importId,
+              mbTilesDbPath: mbTilesDb.name,
+              styleId,
+              tilesetId,
+              port: port1,
+            },
+            { signal: abortSignaler, transferList: [port1] }
+          )
+          .catch((err) => {
+            rej(err)
+          })
+          .finally(cleanup)
 
-      await piscina.run(
-        {
-          dbPath: db.name,
-          importId,
-          mbTilesDbPath: mbTilesDb.name,
-          styleId,
-          tilesetId,
-          port: port1,
-        },
-        { transferList: [port1] }
-      )
+        function handleFirstProgressMessage(message: PortMessage) {
+          if (message.type === 'progress') {
+            port2.off('message', handleFirstProgressMessage)
+            res({ import: { id: message.importId }, tileset })
+          }
+        }
+
+        function cleanup() {
+          clearTimeout(timeoutId)
+          port2.off('message', resetTimeout)
```

Comment:
I _think_ adding this was my attempt to addresss https://github.com/digidem/mapeo-map-server/pull/33/files#r909025866, but not entirely sure if that solved it or not